src/hotspot/share/gc/g1/heapRegionRemSet.cpp

```diff
@@ -145,6 +145,13 @@ void OtherRegionsTable::add_reference(OopOrNarrowOopStar from, uint tid) {
   PerRegionTable* prt = find_region_table(ind, from_hr);
   if (prt == NULL) {
     MutexLocker x(_m, Mutex::_no_safepoint_check_flag);
+
+    // Make sure region hasn't been coarsened by other thread.
```

Comment:
Maybe mentioning this is an intentional re-check under lock, sth like, "Rechecking if the region is coarsened while holding the lock."

Reply:
Yes, thank you!